src/compiler/checker.ts

```diff
@@ -4849,7 +4849,7 @@ namespace ts {
             if (strictNullChecks && declaration.initializer && !(getFalsyFlags(checkDeclarationInitializer(declaration)) & TypeFlags.Undefined)) {
                 type = getTypeWithFacts(type, TypeFacts.NEUndefined);
             }
-            return declaration.initializer && !getEffectiveTypeAnnotationNode(walkUpBindingElementsAndPatterns(declaration)) ?
+            return declaration.initializer && !getEffectiveTypeAnnotationNode(walkUpBindingElementsAndPatterns(declaration)) && !getContextualTypeForVariableLikeDeclaration(walkUpBindingElementsAndPatterns(declaration)) ?

```

Comment:
Isn't the `!getEffectiveTypeAnnotationNode(...)` redundant here? It's the first thing checked in `getContextualTypeForVariableLikeDeclaration`.

Reply:
Mmm yes. I'll remove it.

Reply:
And it's gone~